ui/src/clockface/components/label/Label.tsx

```diff
@@ -59,34 +59,37 @@ class Label extends Component<Props, State> {
     return (
       <div
         className={this.className}
-        onClick={this.handleClick}
         onMouseEnter={this.handleMouseEnter}
         onMouseLeave={this.handleMouseLeave}
         style={this.style}
         title={this.title}
-        data-testid={`${testID} ${name}`}
       >
-        <label>{name}</label>
+        <span
+          className="label--name"
+          onClick={this.handleClick}
+          data-testid={`${testID} ${name}`}
+        >
+          {name}
+        </span>
         {this.deleteButton}
       </div>
     )
   }
 
   private handleClick = (e: MouseEvent<HTMLDivElement>): void => {
-    e.preventDefault()
     const {id, onClick} = this.props
 
     if (onClick) {
+      e.stopPropagation()
+      e.preventDefault()
       onClick(id)
     }
   }
 
   private handleDelete = (): void => {
     const {id, onDelete} = this.props
 
-    if (onDelete) {
-      onDelete(id)
-    }
+    onDelete(id)
```

Comment:
`onDelete` is optional, perhaps this should remain in an `if` statement.

Reply:
I took it out because the button that calls the delete function is also conditionally rendered with the same check, seemed redundant. I can add this guard back in to be extra safe